src/ring_theory/dedekind_domain/ideal.lean

```diff
@@ -781,6 +780,30 @@ begin
     { exact this } },
 end
 
+lemma irreducible_pow_sup (hI : I ≠ ⊥) (hJ : irreducible J) (n : ℕ) :
+  J^n ⊔ I = J^(min ((normalized_factors I).count J) n) :=
+by rw [sup_eq_prod_inf_factors (pow_ne_zero n hJ.ne_zero) hI, ← inf_eq_inter,
+       normalized_factors_of_irreducible_pow hJ, normalize_eq J, repeat_inf, prod_repeat]
+
+lemma irreducible_pow_sup_of_ge (hI : I ≠ ⊥) (hJ : irreducible J) (n : ℕ)
```

Comment:
```suggestion
lemma irreducible_pow_sup_of_le (hI : I ≠ ⊥) (hJ : irreducible J) (n : ℕ)
```
I would call this one `of_le`, and the following `of_ge`, since the exponent is `n`.